### PR TITLE
Vue3 - update workflows ci

### PR DESCRIPTION
**Pull request by jasonlundien**
This PR updates the workflows `ci.yml` file to include more of the matrix style testing including the different versions of Vue and Font Awesome.  

**Reviewers:**
@robmadole --- will you review for accuracy and see if I have missed anything ?

**Dependencies:**
none

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,9 +36,14 @@ jobs:
         node-version: ${{ matrix.node-version }}
     - name: install, build, and test
       run: |
+        npm install -g npm
         npm install
+        npm install --no-save @fortawesome/fontawesome-svg-core@${{ matrix.fontawesome-svg-core }} vue@${{ matrix.vue }}
```
Comment (robmadole) [inline .github/workflows/ci.yml]: @jasonlundien Looks like we are missing the installation of `free-solid-svg-icons`?

Comment (jasonlundien) [reply .github/workflows/ci.yml]: Doooolp !!!  Fixed this up !!!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,9 +36,14 @@ jobs:
         node-version: ${{ matrix.node-version }}
     - name: install, build, and test
       run: |
+        npm install -g npm
         npm install
+        npm install --no-save @fortawesome/fontawesome-svg-core@${{ matrix.fontawesome-svg-core }} @fortawesome/free-solid-svg-icons@${{ matrix.free-solid-svg-icons }} vue@${{ matrix.vue }}
         npm run build
-        npm run test
+        npm install --no-save vue@${{ matrix.vue }}
```
Comment (robmadole) [inline .github/workflows/ci.yml]: @jasonlundien are we installing `vue` twice?

Comment (jasonlundien) [reply .github/workflows/ci.yml]: yes we are, so I will remove this line !

**Comment by robmadole** [comment]
@jasonlundien it's starting to come back to me a bit!

We need to update package.json.

```js
"test": "npm run test.5 && npm run test.6",
    "test.5": "npm --no-save install @fortawesome/fontawesome-svg-core@1.2.x @fortawesome/free-solid-svg-icons@5.x && jest --silent",
    "test.6": "npm --no-save install @fortawesome/fontawesome-svg-core@6.x @fortawesome/free-solid-svg-icons@6.x && jest --silent",
```

We don't need the separate `test.5` `test.6`. We should be able to just run `jest --silent`.

**Pull request merged**